submissions/jelz/src/jelz/core.cljs

```diff
@@ -0,0 +1,116 @@
+(ns ^:figwheel-always jelz.core
+  (:require-macros
+    [reagent.ratom :refer [reaction]])
+  (:require
+    [jelz.view :refer [layout]]
+    [jelz.client :as client]
+    [jelz.helpers :as h]
+    [reagent.core :as reagent]
+    [re-frame.core :refer [register-handler register-sub dispatch]]))
+
+;;-- Handlers ------------------------------------------------------------------
+
+(register-handler
+  :planet-change
+  (fn [db [_ planet]]
+    (dispatch [:danger-check])
+    (assoc db :planet planet)))
+
+(register-handler
+  :jedi-reset
+  (fn [db _]
+    (dispatch [:danger-check])
+    (assoc db :slots (h/get-empty-slots))))
+
+(register-handler
+  :jedi-request-first
+  (fn [db _]
+    (dispatch [:jedi-request (:req-id (first (:slots db))) 3616])
```

Comment:
Instead of 
`(:req-id (first (:slots db)))`
I would personally find it easier to read
`(-> db :slots first :req-id)`

But that's just a personal opinion.  Others may well not agree.  Just noting.